providers/evercrypt/fst/EverCrypt.Poly1305.fst

```diff
@@ -126,17 +126,26 @@ let poly1305 dst src len key =
   let h0 = ST.get () in
   let avx2 = EverCrypt.AutoConfig2.has_avx2 () in
   let avx = EverCrypt.AutoConfig2.has_avx () in
+  let vec256 = EverCrypt.AutoConfig2.has_vec256 () in
+  let vec128 = EverCrypt.AutoConfig2.has_vec128 () in
   let vale = EverCrypt.AutoConfig2.wants_vale () in
 
-  if EverCrypt.TargetConfig.x64 && avx2 then begin
+  if EverCrypt.TargetConfig.evercrypt_can_compile_vec256 && vec256 then begin
     Hacl.Poly1305_256.poly1305_mac dst len src key
 
-  end else if EverCrypt.TargetConfig.x64 && avx then begin
+  end else if EverCrypt.TargetConfig.evercrypt_can_compile_vec128 && vec128 then begin
     Hacl.Poly1305_128.poly1305_mac dst len src key
 
-  end else if EverCrypt.TargetConfig.x64 && vale then begin
-    poly1305_vale dst src len key
+  end else if EverCrypt.TargetConfig.evercrypt_can_compile_vale && vale then begin
 
-  end else begin
+    if EverCrypt.TargetConfig.evercrypt_can_compile_vec128 then begin
```

Comment:
I feel like this test here should reference specific CPU flags

also, wants_vale is perhaps legacy and we should consider deprecating it -- it was useful for tests, but frankly, there's just too many flags now

Reply:
Yes indeed, this only depends on Vale. I'm however not completely sure about what to do for the `wants_vale` functions. I added the `deprecated` attribute, but I don't know if I can completely remove all the occurrences.

Reply:
Frankly I think we should leave this to a further cleanup. Remove: wants_vale, StaticConfig, and all of that stuff that predates the nice configuration infrastructure that you instated and materialized in this PR. If you start this now, you'll have to fix a bunch of tests, which is not very fun.